### PR TITLE
Bugfix/emptyline no whitespace

### PR DESCRIPTION
**Pull request by jspaezp**
Fixes: https://github.com/DanielNoord/pydocstringformatter/issues/159

### CODE REVIEW

File: tests/data/format/no_whitespace_stripper/module_docstring.py.out [hunk 1]
```diff
@@ -0,0 +1,18 @@
+"""  A multi-line  .
```
Comment (Pierre-Sassoulas) [inline tests/data/format/no_whitespace_stripper/module_docstring.py.out:1]: I wonder if we should rstrip the end of the line before adding the final dot. Thought @DanielNoord ?

**Review by Pierre-Sassoulas — COMMENTED** [review]
Looks very nice, a straightforward fix and a good accompanying test, thank you.

File: tests/data/format/no_whitespace_stripper/module_docstring.py.out (hunk 1 above)
Comment (jspaezp) [reply tests/data/format/no_whitespace_stripper/module_docstring.py.out:1]: I was thinking the same thing while writing the test, but I figured at the end of the day the argument is called `--no-strip-whitespace` .... so I figured it was consistent in its behavior.

Maybe down the road we could change the argument such that there is the possibility of only removing trailing whitespace, instead of always removing both trailing and starting.

LMK what you think

File: pydocstringformatter/_formatting/_utils.py [hunk 2]
```diff
@@ -6,4 +6,6 @@ def is_rst_title(summary: str) -> bool:
     """Check if the second line of a summary is one recurring character."""
     # If second line is one recurring character we're dealing with a rst title
     last_line = summary.splitlines()[-1].lstrip()
+    if len(last_line) < 1:
```
Comment (DanielNoord) [inline pydocstringformatter/_formatting/_utils.py:9]:
```suggestion
    if not last_line:
```

This should also work and save the call to `len`!

File: tests/data/format/no_whitespace_stripper/module_docstring.py.out (hunk 1 above)
Comment (DanielNoord) [reply tests/data/format/no_whitespace_stripper/module_docstring.py.out:1]: Yeah agreed.

I think if you're letting a tool put periods in your docstrings automatically you'll likely also let that tool remove whitespaces. So they probably won't be without the other very often. And in those cases where people do I think it is good to offer the flexibility.

File: tests/data/format/no_whitespace_stripper/module_docstring.py.out [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+docstring  
+   
+"""
+
+"""  My docstring.
+
+ 
+    
+   My indented section  
+
```
Comment (DanielNoord) [inline tests/data/format/no_whitespace_stripper/module_docstring.py.out:17]: This seems to be a very weird edge case. Shouldn't L17 have as many whitespaces as L12 has in the original file?

Comment (DanielNoord) [inline tests/data/format/no_whitespace_stripper/module_docstring.py.out:17]: This seems to be a very weird edge case. Shouldn't L17 have as many whitespaces as L12 has in the original file?

Comment (jspaezp) [reply tests/data/format/no_whitespace_stripper/module_docstring.py.out:17]: I think It does ...

(screenshot from my vscode just to render the whitespace ...)
image

Comment (jspaezp) [reply tests/data/format/no_whitespace_stripper/module_docstring.py.out:17]: I believe it does (at least when looking at it in vscode to render the whitespace)

File: pydocstringformatter/_formatting/_utils.py [hunk 4]
```diff
@@ -5,5 +5,7 @@
 def is_rst_title(summary: str) -> bool:
     """Check if the second line of a summary is one recurring character."""
     # If second line is one recurring character we're dealing with a rst title
-    last_line = summary.splitlines()[-1].lstrip()
+
```
Comment (DanielNoord) [inline pydocstringformatter/_formatting/_utils.py:8]:
```suggestion
```

File: tests/data/format/no_whitespace_stripper/module_docstring.py.out (hunk 3 above)
Comment (DanielNoord) [reply tests/data/format/no_whitespace_stripper/module_docstring.py.out:17]: Ah yeah, it does indeed. I should have checked the raw file myself.

**Review by DanielNoord — APPROVED** [review]

**Review by DanielNoord — APPROVED** [review]
